src/code_80A2C3C.c

```diff
@@ -0,0 +1,34 @@
+#include "global.h"
+#include "bmsave.h"
+
+extern u8 LoadAndVerifySecureHeaderSW(struct SecureSaveHeader* buf);
+extern void SaveSecureHeader(struct SecureSaveHeader* header);
+
+// unitID is CharacterData.number. Should I just use "number" as variable name?
+void SetSomeUnitStatThingUnlockMaybeIdk(s32 unitID, struct SecureSaveHeader* buf)
+{
+  s32 boolLoadedSecureHeader = 0, groupID, unitIDInGroup;
+  u8* loc;
+  struct SecureSaveHeader tmp_header;
+  
+  if (unitID > 256) {
+    return;
+  }
+  if (buf == NULL) {
+    buf = &tmp_header;
+    LoadAndVerifySecureHeaderSW(buf);
+    boolLoadedSecureHeader = 1;
+  }
+  
+  groupID = unitID >> 3; // Units are divided into 32 groups containing 8 units each.
+  loc = &buf->unk40[0];
+  loc += groupID;
+  unitIDInGroup = unitID & 0x7; // First three bits.
```

Comment:
this can just be
```c
buf->unk40[unitID >> 3] |= 1 << (unitID & 7);
```
also somebody rename fields in this struct, [most of those are known](https://github.com/StanHash/DOC/blob/master/SaveMetadata.txt#L35).